src/cryptonote_basic/difficulty.cpp

```diff
@@ -162,4 +162,81 @@ namespace cryptonote {
     return (low + time_span - 1) / time_span;
   }
 
+#if defined(_MSC_VER)
+#ifdef max
+#undef max
+#endif
+#endif
+
+  const difficulty_type max64bit(std::numeric_limits<std::uint64_t>::max());
+  const boost::multiprecision::uint256_t max128bit(std::numeric_limits<boost::multiprecision::uint128_t>::max());
+  const boost::multiprecision::uint512_t max256bit(std::numeric_limits<boost::multiprecision::uint256_t>::max());
+
+#define FORCE_FULL_128_BITS
+
+  bool check_hash_128(const crypto::hash &hash, difficulty_type difficulty) {
+#ifndef FORCE_FULL_128_BITS
+    // fast check
+    if (((const uint64_t *) &hash)[3] > 0)
+      return false;
```

Comment:
No, because if difficulty < 2^64 it does not go there, it goes through check_hash_64. I'll add a check for difficulty here though, in case it gets called directly.

Reply:
Ah I see.